odoo/tools/config.py

```diff
@@ -61,6 +61,13 @@ def _deduplicate_loggers(loggers):
         for logger, level in dict(it.split(':') for it in loggers).items()
     )
 
+def parse_pico_lang(test_tags):
+    # parsing the test_tags pico language for selecting or deselecting tests
+    pico_tags = {t.strip() for t in test_tags.split(',') if t.strip() != ''}
+    no_test_tags = {t[1:] for t in pico_tags if t.startswith('-')}
+    to_test_tags = {t.replace('+', '') for t in pico_tags if not t.startswith('-')}
+    return (to_test_tags, no_test_tags)
+
```

Comment:
Not sure that it's the right place where to put this function.
What do you thing @xmo-odoo ?

Reply:
I suggest to use a class for wrapping the test selector tags.
The class would implement both the spec parser and the check on test classes.
```python
class TestTags(object):
    """ Test selector based on tags. """

    def __init__(self, spec):
        """ Parse the spec to determine tags to include and exclude. """
        self.include = set()
        self.exclude = set()
        for tag in spec.split(','):
            if tag.startswith('-'):
                self.exclude.add(tag[1:])
            else:
                self.include.add(tag[1:] if tag.startswith('+') else tag)

    def __call__(self, arg):
        """ Return whether ``arg`` matches the specification: it must have at
            least one tag in ``self.include`` and none in ``self.exclude``.
        """
        tags = getattr(arg, 'test_tags', ('standard',))
        return not self.include.isdisjoint(tags) and self.exclude.isdisjoint(tags)
```
It would avoid the need for extra elements in `config`, and provides a straight API for filtering tests:
```python
test_tags = TestTags(tools.config['test-tags'])
...
suite = unittest.TestSuite(t for t in tests if position(t) and test_tags(t))
```